cpp/src/arrow/buffer_builder.h

```diff
@@ -292,6 +292,11 @@ class TypedBufferBuilder<bool> {
     return Status::OK();
   }
 
+  void Forward(int64_t num_elements) {
```

Comment:
This will leave some entries undefined unless they're written manually elsewhere, which I don't think happens here.
I don't think this method is needed, instead you should just `Append(true)` or `Append(length, true)`, IMHO.

Reply:
See comments above~

Reply:
This method was removed.